src/Module/Settings/Connectors.php

```diff
@@ -0,0 +1,247 @@
+			$mail_pubmail      =                 $request['mail_pubmail'] ?? '';
+
+			if (!$this->database->exists('mailacct', ['uid' => $this->session->getLocalUserId()])) {
+				$this->database->insert('mailacct', ['uid' => $this->session->getLocalUserId()]);
+			}
+
+			if (strlen($mail_pass)) {
+				$pass = '';
+				openssl_public_encrypt($mail_pass, $pass, $user['pubkey']);
+				$this->database->update('mailacct', ['pass' => bin2hex($pass)], ['uid' => $this->session->getLocalUserId()]);
+			}
+
+			$r = $this->database->update('mailacct', [
+				'server'       => $mail_server,
+				'port'         => $mail_port,
+				'ssltype'      => $mail_ssl,
+				'user'         => $mail_user,
+				'action'       => $mail_action,
+				'movetofolder' => $mail_movetofolder,
+				'mailbox'      => 'INBOX',
+				'reply_to'     => $mail_replyto,
+				'pubmail'      => $mail_pubmail
+			], ['uid' => $this->session->getLocalUserId()]);
+
+			$this->logger->debug('updating mailaccount', ['response' => $r]);
+			$mailacct = $this->database->selectFirst('mailacct', [], ['uid' => $this->session->getLocalUserId()]);
+			if ($this->database->isResult($mailacct)) {
+				if (strlen($mailacct['server'])) {
+					$dcrpass = '';
+					openssl_private_decrypt(hex2bin($mailacct['pass']), $dcrpass, $user['prvkey']);
+					$mbox = Email::connect(Email::constructMailboxName($mailacct), $mail_user, $dcrpass);
+					unset($dcrpass);
+					if (!$mbox) {
+						$this->systemMessages->addNotice($this->t('Failed to connect with email account using the settings provided.'));
+					}
+				}
+			}
+		}
+
+		Hook::callAll('connector_settings_post', $request);
+		$this->baseUrl->redirect($this->args->getQueryString());
+	}
+
+	protected function content(array $request = []): string
+	{
+		parent::content($request);
+
+		$accept_only_sharer      =  intval($this->pconfig->get($this->session->getLocalUserId(), 'system', 'accept_only_sharer'));
+		$enable_cw               = !intval($this->pconfig->get($this->session->getLocalUserId(), 'system', 'disable_cw'));
+		$enable_smart_shortening = !intval($this->pconfig->get($this->session->getLocalUserId(), 'system', 'no_intelligent_shortening'));
+		$simple_shortening       =  intval($this->pconfig->get($this->session->getLocalUserId(), 'system', 'simple_shortening'));
+		$attach_link_title       =  intval($this->pconfig->get($this->session->getLocalUserId(), 'system', 'attach_link_title'));
+		$legacy_contact          =         $this->pconfig->get($this->session->getLocalUserId(), 'ostatus', 'legacy_contact');
+
+		if (!empty($legacy_contact)) {
+			$this->baseUrl->redirect('ostatus_subscribe?url=' . urlencode($legacy_contact));
+		}
+
+		$connector_settings_forms = [];
+		foreach ($this->database->selectToArray('hook', ['file', 'function'], ['hook' => 'connector_settings']) as $hook) {
+			$data = [];
+			Hook::callSingle(DI::app(), 'connector_settings', [$hook['file'], $hook['function']], $data);
```

Comment:
```suggestion
			Hook::callSingle($this->app, 'connector_settings', [$hook['file'], $hook['function']], $data);
```
And add `App` as new dependency :)
--> With a new Addon system, this enforced App parameter should vanish ..